tests/h/views/accounts_test.py

```diff
@@ -688,97 +688,71 @@ def routes(self, pyramid_config):
         pyramid_config.add_route("account", "/my/account")
 
 
-@pytest.mark.usefixtures("pyramid_config", "routes", "subscriptions_model")
 class TestNotificationsController:
-    def test_get_sets_subscriptions_data_in_form(
-        self, form_validating_to, pyramid_config, pyramid_request, subscriptions_model
-    ):
-        pyramid_config.testing_securitypolicy("fiona")
+    def test_get(self, controller, authenticated_user, subscriptions_model):
         subscriptions_model.get_subscriptions_for_uri.return_value = [
             FakeSubscription("reply", True),
             FakeSubscription("foo", False),
         ]
-        controller = views.NotificationsController(pyramid_request)
-        controller.form = form_validating_to({})
 
-        controller.get()
+        result = controller.get()
 
         controller.form.set_appstruct.assert_called_once_with(
             {"notifications": {"reply"}}
         )
-
-    def test_it_does_not_render_form_if_user_has_no_email_address(
-        self, factories, form_validating_to, pyramid_request
-    ):
-        controller = views.NotificationsController(pyramid_request)
-        controller.form = form_validating_to({})
-        pyramid_request.user = factories.User(username="cara")
-        pyramid_request.user.email = None
-
-        result = controller.get()
-
-        assert "form" not in result
-        assert result["user_has_email_address"] is None
-
-    def test_it_renders_form_if_user_has_email_address(
-        self, factories, form_validating_to, pyramid_request
-    ):
-        controller = views.NotificationsController(pyramid_request)
-        controller.form = form_validating_to({})
-        pyramid_request.user = factories.User(username="janedoe")
-
-        result = controller.get()
```

Comment:
Some of the testing style where we separate assertions about what the method calls and what it returns etc. Sometimes you see a different test for each thing the methods calls.

This causes the setup to be duplicated over and over and makes the flow hard to follow.